yt/utilities/io_handler.py

```diff
@@ -292,30 +292,33 @@ def _read_particle_selection(
 
     def _read_particle_fields(self, chunks, ptf, selector):
         # Now we have all the sizes, and we can allocate
-        data_files = set()
-        for chunk in chunks:
-            for obj in chunk.objs:
-                data_files.update(obj.data_files)
-        for data_file in sorted(data_files, key=lambda x: (x.filename, x.start)):
+        for data_file in self._sorted_chunk_iterator(chunks):
             data_file_data = self._read_particle_data_file(data_file, ptf, selector)
             # temporary trickery so it's still an iterator, need to adjust
             # the io_handler.BaseIOHandler.read_particle_selection() method
             # to not use an iterator.
             yield from data_file_data.items()
 
-
-# As a note: we don't *actually* want this to be how it is forever.  There's no
-# reason we need to have the fluid and particle IO handlers separated.  But,
-# for keeping track of which frontend is which, this is a useful abstraction.
-class BaseParticleIOHandler(BaseIOHandler):
-    def _sorted_chunk_iterator(self, chunks):
+    def _data_files_set(self, chunks) -> set:
+        # returns a set of datafiles from across chunks
         chunks = list(chunks)
         data_files = set()
         for chunk in chunks:
             for obj in chunk.objs:
                 data_files.update(obj.data_files)
+        return data_files
+
+    def _sorted_chunk_iterator(self, chunks):
+        # yields the data files contained in a chunks iterator sorted by
+        # filename and starting position with each file
+        data_files = self._data_files_set(chunks)
         yield from sorted(data_files, key=lambda x: (x.filename, x.start))
 
+
+# As a note: we don't *actually* want this to be how it is forever.  There's no
+# reason we need to have the fluid and particle IO handlers separated.  But,
+# for keeping track of which frontend is which, this is a useful abstraction.
```

Comment:
since moving the comment is going to make git blame less useful, can you link to the original PR this was added ?
```suggestion
# for keeping track of which frontend is which, this is a useful abstraction.
# see https://github.com/yt-project/yt/pull/3526
```

Reply:
updated in the latest commit